griptape/engines/eval/eval_engine.py

```diff
@@ -0,0 +1,137 @@
+from griptape.engines import BaseEvalEngine
+from griptape.mixins.serializable_mixin import SerializableMixin
+from griptape.rules import JsonSchemaRule
+from griptape.utils import J2
+
+if TYPE_CHECKING:
+    from griptape.drivers import BasePromptDriver
+
+STEPS_SCHEMA = schema.Schema(
+    {
+        "steps": [str],
+    }
+)
+
+RESULTS_SCHEMA = schema.Schema(
+    {
+        "score": float,
+        "reason": str,
+    }
+)
+
+
+@define(kw_only=True)
+class EvalEngine(BaseEvalEngine, SerializableMixin):
+    id: str = field(default=Factory(lambda: uuid.uuid4().hex), kw_only=True, metadata={"serializable": True})
+    name: str = field(
+        default=Factory(lambda self: self.id, takes_self=True),
+        metadata={"serializable": True},
+    )
+    criteria: Optional[str] = field(default=None, metadata={"serializable": True})
+    evaluation_steps: Optional[list[str]] = field(default=None, metadata={"serializable": True})
+    prompt_driver: BasePromptDriver = field(default=Factory(lambda: Defaults.drivers_config.prompt_driver))
+    generate_steps_system_template: J2 = field(default=Factory(lambda: J2("engines/eval/steps/system.j2")))
+    generate_steps_user_template: J2 = field(default=Factory(lambda: J2("engines/eval/steps/user.j2")))
+    generate_results_system_template: J2 = field(default=Factory(lambda: J2("engines/eval/results/system.j2")))
+    generate_results_user_template: J2 = field(default=Factory(lambda: J2("engines/eval/results/user.j2")))
+
+    @criteria.validator  # pyright: ignore[reportAttributeAccessIssue, reportOptionalMemberAccess]
+    def validate_criteria(self, _: Attribute, value: Optional[str]) -> None:
+        if value is None:
+            if self.evaluation_steps is None:
+                raise ValueError("either criteria or evaluation_steps must be specified")
+            return
+
+        if self.evaluation_steps is not None:
+            raise ValueError("can't have both criteria and evaluation_steps specified")
+
+        if not value:
+            raise ValueError("criteria must not be empty")
+
+    @evaluation_steps.validator  # pyright: ignore[reportAttributeAccessIssue, reportOptionalMemberAccess]
+    def validate_evaluation_steps(self, _: Attribute, value: Optional[list[str]]) -> None:
+        if value is None:
+            if self.criteria is None:
+                raise ValueError("either evaluation_steps or criteria must be specified")
+            return
+
+        if self.criteria is not None:
+            raise ValueError("can't have both evaluation_steps and criteria specified")
+
+        if not value:
+            raise ValueError("evaluation_steps must not be empty")
+
+    def evaluate(self, input: str, actual_output: str, **kwargs) -> tuple[float, str]:  # noqa: A002
+        evaluation_params = {
+            key.replace("_", " ").title(): value
+            for key, value in {"input": input, "actual_output": actual_output, **kwargs}.items()
+        }
```

Comment:
Why do you need to transform the key this way? Why pass kwargs as params? (don't see it used in test or example)

Reply:
Probably don't need to, just wanted to make it look less pythonic to the LLM. As if a user was explaining it in plain english.